lib/SPIRVProducerPass.cpp

```diff
@@ -3899,21 +3896,22 @@ void SPIRVProducerPass::GenerateInstruction(Instruction &I) {
     }
 
     if (Opcode == spv::OpPtrAccessChain) {
-      // Do we need to generate ArrayStride?  Check against the GEP result type
-      // rather than the pointer type of the base because when indexing into
-      // an OpenCL program-scope constant, we'll swap out the LLVM base pointer
-      // for something else in the SPIR-V.
-      // E.g. see test/PointerAccessChain/pointer_index_is_constant_1.cl
+      // Shader validation in the SPIR-V spec requires that the base pointer to
+      // OpPtrAccessChain (in StorageBuffer storage class) be decorated with
+      // ArrayStride.
       auto address_space = ResultType->getAddressSpace();
       setVariablePointersCapabilities(address_space);
       switch (GetStorageClass(address_space)) {
       case spv::StorageClassStorageBuffer:
-      case spv::StorageClassUniform:
         // Save the need to generate an ArrayStride decoration.  But defer
         // generation until later, so we only make one decoration.
-        getTypesNeedingArrayStride().insert(ResultType);
+        getTypesNeedingArrayStride().insert(GEP->getPointerOperandType());
+        break;
+      case spv::StorageClassWorkgroup:
```

Comment:
Just curious: do we ever generate `OpPtrAccessChain` for `Workgroup` storage class (presumably we don't have test coverage for this)?